src/merkle_sum_tree/utils/csv_parser.rs

```diff
@@ -5,12 +5,18 @@ use std::error::Error;
 use std::fs::File;
 use std::path::Path;
 
-#[derive(Debug, Deserialize)]
-struct CsvEntry {
-    username: String,
-    balance: Vec<String>,
+macro_rules! csv_entry {
```

Comment:
The whole `csv_parser` doesn't seem very modular to me. I believe that if I want to use multi_assets 3, even if I generate the correct parameters, there's no way to parse it without manually modifying the csv_parser function.

Reply:
This would require a ; separator between "user" and "balances" which becomes a list of a length N_ASSETS.

Reply:
Nice! I also thought about using the macro here, `csv_parser`.
I believe we should consider refactoring this after we've defined the API, as it closely ties in with API.
If everyone agrees, I'd be happy to create an issue for this refactoring task.

Reply:
I believe this is already resolved by putting the array inside the 'balance' in CSV - this way the CSV should only have 'user' and 'balance' columns, but the records in 'balance' are expected to have N_ASSETS elements. Columns are semicolon-separated, and the balance values are comma-separated.

Reply:
I understand, it's a very straightforward solution! I like it.